swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -1,5 +1,1304 @@
 """
-Particle generation code.
+Generate SPH initial conditions for SPH simulations iteratively for a given density function following Arth et al.
+2019 (https://arxiv.org/abs/1907.11250).
 
-TBD
 """
+import numpy as np
+import unyt
+from math import erf
+from typing import Union
+import warnings
+
+from .IC_kernel import get_kernel_data
+from swiftsimio.optional_packages import KDTree, TREE_AVAILABLE
+from swiftsimio import Writer
+
+
+class RunParams(object):
+    r"""
+    A class to store iteration runtime parameters. Before running the iteration,
+    these parameters are intended to be direcly accessed by the user.
+
+
+    Attributes
+    ---------------
+    self.iter_max: int
```

Comment:
You don't need `self.` here, just `iter_max`.

Reply:
PyCharm was complaining about not having the `self` though

Reply:
The problem is that in our ReadTheDocs output it ends up looking like this: 
<img width="765" alt="image" src="https://user-images.githubusercontent.com/7839515/88916557-2f448580-d25e-11ea-88ed-f21748e7acbe.png">
which just has unecessary self's.

Reply:
Done.